src/Orleans/Async/AsyncLock.cs

```diff
@@ -35,15 +35,16 @@ namespace Orleans
     /// }
     /// </code>
     /// </example>
-    /// 
+    ///
     /// We decided to keep the implemention simple and mimic the semantics of a regular mutex as much as possible.
-    /// 1) AsyncLock is NOT IDisposable, since we don't want to give the developer an option to erraneously manualy dispose the lock 
+    /// 1) AsyncLock is NOT IDisposable, since we don't want to give the developer an option to erraneously manualy dispose the lock
     /// while there may be some unreleased LockReleasers.
-    /// 2) AsyncLock does NOT have to implement the Finalizer function. The underlying resource of SemaphoreSlim will be eventually released by the .NET, 
+    /// 2) AsyncLock does NOT have to implement the Finalizer function. The underlying resource of SemaphoreSlim will be eventually released by the .NET,
     /// when SemaphoreSlim is finalized. Having finalizer for AsyncLock will not speed it up.
     /// 3) LockReleaser is IDisposable to implement the "using" pattern.
     /// 4) LockReleaser does NOT have to implement the Finalizer function. If users forget to Dispose the LockReleaser (analagous to forgetting to release a mutex)
-    /// the AsyncLock wil remain locked, which may potentialy cause deadlock. This is OK, since these are the exact regular mutex semantics - if one forgets to unlcok the mutex, it stays locked. 
+    /// the AsyncLock wil remain locked, which may potentialy cause deadlock. This is OK, since these are the exact regular mutex semantics - if one forgets to unlock the mutex, it stays locked.
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1001:TypesThatOwnDisposableFieldsShouldBeDisposable")]
```

Comment:
This is a stop gap solution to avoid going down the rabbit hole of classes that use `AsyncLock` and making them all `IDisposable`, right? I think that's fine. We can open an issue to track removal of it. Low priority IMO.